lib/opam_cmd.ml

```diff
@@ -107,22 +61,22 @@ let classify_package ~package ~dev_repo ~archive ~pins () =
           (`Virtual, None)
       | Some archive -> (
           let uri = Uri.of_string dev_repo in
-          let tag = tag_from_archive archive in
+          let git_ref = Git_ref.from_archive_url archive in
           Logs.debug (fun l ->
-              l "Mapped %s -> %s" archive (match tag with None -> "??" | Some v -> v) );
+              l "Mapped %s -> %s" archive (match git_ref with None -> "??" | Some v -> Git_ref.to_string v) );
           match Uri.host uri with
           | Some "github.com" -> (
             match String.cuts ~empty:false ~sep:"/" (Uri.path uri) with
             | [ user; repo ] ->
-                let repo = strip_ext repo in
-                (`Github (user, repo), tag)
+                let repo = Duniverse_std.File.strip_ext repo in
```

Comment:
Github repos can contain dots and the `.git` extension is optional. This is a bug.